lxc/remote.go

```diff
@@ -150,6 +150,148 @@ func (c *cmdRemoteAdd) findProject(d lxd.InstanceServer, project string) (string
 	return project, nil
 }
 
+func (c *cmdRemoteAdd) parseURL(addr string) (string, error) {
```

Comment:
This looks way overly complex.

The addresses in a join token can only ever be https, so we don't really need all of the checks around scheme and unix and whatnot here.

You can pretty much just prefix the address with `https://` and call it done.